Add aarch64 target for darwin desktops

skhamis opened the pull request:
fixes #77 

Turned out to be a very small patch! Tested with both x86_64 and arm64 JDKs and both seem to compile nicely (the a-s issue mentioned in https://github.com/mozilla/rust-android-gradle/issues/77) . The only issue is I don't necessarily know the best way to handle the deprecation of darwin to officially split darwin to use the {-x86_64, -aarch64}. Throwing this up for early visibility....

skhamis commented:
@ncalexan What do you think of going the darwin + darwin-{x86_64, aarch64} vs just removing the darwin target (breaking change) and only have the darwin with the modifiers? Curious your thoughts!

ncalexan commented on plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt:
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
                 "<compilerTriple>",
                 "<binutilsTriple>",
                 "desktop/linux-x86-64"),
+        // TODO: Best way to handle deprecation here?
```
Fold the third commit into this one, please.

ncalexan commented on plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt:
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
+                "x86_64-apple-darwin",
+                "<compilerTriple>",
+                "<binutilsTriple>",
+                "desktop/darwin"),
```
These need to go elsewhere, surely.  Just `desktop/darwin-x86-64`.

ncalexan commented on plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt:
```diff
@@ -25,12 +25,25 @@ val toolchains = listOf(
+                "aarch64-apple-darwin",
+                "<compilerTriple>",
+                "<binutilsTriple>",
+                "desktop/darwin"),
```
Ditto.

ncalexan commented on README.md:
```diff
@@ -160,8 +161,17 @@ cargo {
-`'win32-x86-64-gnu'`, and `'win32-x86-64-msvc'` (Desktop).
+Valid targets for **Android** are:
+
+>`'arm'`, `'arm64'`, `'x86'`, `'x86_64'`
```
This is an improvement, but let's make it way easier to copy-paste: maybe just a block with each entry on a line, or a list?

ncalexan commented on README.md:
```diff
@@ -160,8 +161,17 @@ cargo {
+
+Valid targets for **Desktop** are:
+> `'linux-x86-64'`,
+ `'darwin-x86-64' (Intel)`,
```
Drop the thing in parentheses that won't be copy-pasted.

ncalexan commented on README.md:
````diff
@@ -481,6 +491,19 @@ $ ls -al samples/maven-repo/org/mozilla/rust-android-gradle/org.mozilla.rust-and
 ```
 
+# Testing Local changes
+After running `./gradlew publish` -- An easy way to locally test changes made in this plugin is to simply add this to your project's `settings.gradle`:
````
You needn't run `./gradlew publish` to use this, in fact.  The plugin will be built (and updated!) as needed.

ncalexan commented on README.md:
````diff
@@ -526,7 +549,7 @@ $ ls -al samples/library/build//outputs/aar/library-debug.aar
+```gradle
 buildscript {
     repositories {
         maven {
````
Bump the `samples/maven-repo` below to `build/local-repo` while you're here, please.

ncalexan replied:
And the version number.

ncalexan reviewed (approved):
Just nits -- everything looks fine.  I'll land this squashed if you prefer to not squash it yourself.  I would like to see it rebased onto https://github.com/mozilla/rust-android-gradle/pull/81 once those tests are green in automation, the new tests updated to include the new Darwin targets and locations, and repushed to be sure we've got this correct.  If you have time, please do it; otherwise, I'll try to get to it shortly.

Thanks, @skhamis!

skhamis commented:
@ncalexan Thanks for the review! I have applied all requested nits/changes and rebased ontop of https://github.com/mozilla/rust-android-gradle/pull/81 for those CI runs. I also squashed and bumped the plugin version -- let me know if I missed anything! Should look good once the CIs pass!

ncalexan commented:
I landed this manually onto `master`: c4bf58464af845fb97bbf7b15a5f4603f3af9dca.

Sorry for the unusual landing; I had some CI testing to work through.  Thanks for pushing this forward, @skhamis!

ncalexan commented:
Oh -- I cut v0.9.1 with this change as well.  Please test and make sure it works for y'all!
